packages/ui/src/components/search-files.tsx

```diff
@@ -15,39 +15,42 @@ import {
 
 interface SearchFilesProps {
   navigateToFile: (file: string) => void
-  filesList: string[] | undefined
+  filesList?: string[]
 }
 
 export const SearchFiles = ({ navigateToFile, filesList }: SearchFilesProps) => {
   const [isOpen, setIsOpen] = useState(false)
   const [query, setQuery] = useState('')
 
-  const handleInputChange = useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+  const handleInputChange = useCallback((event: ChangeEvent<HTMLInputElement>) => {
     const value = event.target.value
     setQuery(value)
     setIsOpen(value !== '')
   }, [])
 
   const filteredFiles = useMemo(() => {
-    return filesList?.filter(file => file.toLowerCase().includes(query.toLowerCase()))
+    if (!filesList) return [] as string[]
+
+    return filesList.filter(file => file.toLowerCase().includes(query.toLowerCase()))
   }, [query, filesList])
 
   const fileText = useCallback(
```

Comment:
This being a callback won't actually help performance at all. `useCallback` is great when you need to stabilise a reference to a function when passing between components (or to a hook), but doesn't offer anything in this case.

If you want to optimise this, then you can preprocess the files list in a `useMemo` so that they start off in the right format and only change if the filesList changes.